src/mscorlib/corefx/System/Security/SecureString.Windows.cs

```diff
@@ -0,0 +1,268 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+namespace System.Security
+{
+    public sealed partial class SecureString
+    {
+        internal SecureString(SecureString str)
+        {
+            BCLDebug.Assert(str != null, "Expected non-null SecureString");
+            BCLDebug.Assert(str._buffer != null, "Expected other SecureString's buffer to be non-null");
+            BCLDebug.Assert(str._encrypted, "Expected to be used only on encrypted SecureStrings");
+
+            AllocateBuffer(str._buffer.Length);
+            SafeBSTRHandle.Copy(str._buffer, _buffer, str._buffer.Length * sizeof(char));
+
+            _decryptedLength = str._decryptedLength;
+            _encrypted = str._encrypted;
+        }
+
+        private unsafe void InitializeSecureString(char* value, int length)
+        {
+            BCLDebug.Assert(length >= 0, $"Expected non-negative length, got {length}");
```

Comment:
Ah - yes! When I originally ported this, your change was not there :).

Will change it!